src/data-api/data-api-client.js

```diff
@@ -0,0 +1,755 @@
+/* eslint-disable func-names */
+/* eslint-disable no-nested-ternary */
+/* eslint-disable spaced-comment */
+/* eslint-disable no-await-in-loop */
+/* eslint-disable no-param-reassign */
+
+/*
+ * This module provides a simplified interface into the Aurora Serverless
+ * Data API by abstracting away the notion of field values.
+ *
+ * This module is mostly copied over from https://github.com/markusahlstrand/knex-data-api-client
+ * and added support for AWS SDK V3 and it's new structure changes
+ *
+ * More detail regarding the Aurora Serverless Data APIcan be found here:
+ * https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/data-api.html
+ *
+ */
+
+const {
+  RDSDataClient,
+  ExecuteStatementCommand,
+  BatchExecuteStatementCommand,
+  BeginTransactionCommand,
+  CommitTransactionCommand,
+  RollbackTransactionCommand
+} = require('@aws-sdk/client-rds-data');
+
+// Supported value types in the Data API
+const supportedTypes = [
+  'arrayValue',
+  'blobValue',
+  'booleanValue',
+  'doubleValue',
+  'isNull',
+  'longValue',
+  'stringValue',
+  'structValue'
+];
+
+/********************************************************************/
+/**  PRIVATE METHODS                                               **/
+/********************************************************************/
+
+// Simple error function
+const error = (...err) => {
+  throw Error(...err);
+};
+
+// Picked from sqlString module to reduce the dependency chain
+function escapeId(val, forbidQualified) {
+  const ID_GLOBAL_REGEXP = /`/g;
+  const QUAL_GLOBAL_REGEXP = /\./g;
+  if (Array.isArray(val)) {
+    let sql = '';
+
+    for (let i = 0; i < val.length; i++) {
+      sql += (i === 0 ? '' : ', ') + escapeId(val[i], forbidQualified);
+    }
+
+    return sql;
+  } if (forbidQualified) {
+    return `\`${  String(val).replace(ID_GLOBAL_REGEXP, '``')  }\``;
+  }
+    return `\`${  String(val).replace(ID_GLOBAL_REGEXP, '``').replace(QUAL_GLOBAL_REGEXP, '`.`')  }\``;
+
+};
+
+// Parse SQL statement from provided arguments
+const parseSQL = (args) =>
+  typeof args[0] === 'string'
+    ? args[0]
+    : typeof args[0] === 'object' && typeof args[0].sql === 'string'
+      ? args[0].sql
+      : error(`No 'sql' statement provided.`);
+
+// Parse the parameters from provided arguments
+const parseParams = args =>
+  Array.isArray(args[0].parameters)
+    ? args[0].parameters
+    : typeof args[0].parameters === 'object'
+      ? [args[0].parameters]
+      : Array.isArray(args[1])
+        ? args[1]
+        : typeof args[1] === 'object'
+          ? [args[1]]
+          : args[0].parameters
+            ? error("'parameters' must be an object or array")
```

Comment:
```suggestion
            ? error('parameters must be an object or array')
```
Keep it consistent?
Either change this or the below... If you change this then check line 99, 109, 120, 126and 130 as well

Reply:
Changed and made consistant by keeping quoted values everywhere.